dotnet/src/SemanticKernel.Abstractions/Memory/IMemoryRecordService{TKey,TDataModel}.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Memory;
+
+/// <summary>
+/// An interface for adding, updating, deleting and retrieving records from a memory store.
```

Comment:
- Should this abstraction be an API for interacting with vector data stores only rather than the more generic memory store?
- Should we restrict what we store to always having associated Vector field(s)?

Reply:
I think the definition of a vector store is becoming more difficult since all databases and indexes seem to be adding vector support right now.  Requiring users to specify a vector field is also tricky, since there may be cases where the embedding is generated in the service from a provided field, and on query the service again generates an embedding from the query text, does the comparison and returns the text.  The client may never have to do anything with vectors.
I think therefore Memory makes sense in that it restricts the use cases to memory access.

Reply:
Still getting input from various people on this as well, so let's keep iterating on this.